patches/0004-Use-crypto-backends.patch

```diff
@@ -379,19 +380,164 @@ index 33942467784ad3..0282ffa9fa23c8 100644
  	"crypto/internal/cryptotest"
  	fipsaes "crypto/internal/fips140/aes"
  	"encoding/hex"
+diff --git a/src/crypto/cipher/gcm.go b/src/crypto/cipher/gcm.go
+index 5580f96d55a0fb..cb6cc900fde083 100644
+--- a/src/crypto/cipher/gcm.go
++++ b/src/crypto/cipher/gcm.go
+@@ -93,7 +93,14 @@ func newGCM(cipher Block, nonceSize, tagSize int) (AEAD, error) {
+ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ 	c, ok := cipher.(*aes.Block)
+ 	if !ok {
+-		return nil, errors.New("cipher: NewGCMWithRandomNonce requires aes.Block")
++		if cipher, ok := cipher.(gcmAble); ok {
++			g, err := cipher.NewGCM(gcmStandardNonceSize, gcmTagSize)
++			if err != nil {
++				return nil, err
++			}
++
++			return gcmWithRandomNonce{g}, nil
++		}
+ 	}
+ 	g, err := gcm.New(c, gcmStandardNonceSize, gcmTagSize)
+ 	if err != nil {
+@@ -103,7 +110,7 @@ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ }
+ 
+ type gcmWithRandomNonce struct {
+-	*gcm.GCM
++	aead AEAD
+ }
+ 
+ func (g gcmWithRandomNonce) NonceSize() int {
+@@ -157,7 +164,14 @@ func (g gcmWithRandomNonce) Seal(dst, nonce, plaintext, additionalData []byte) [
+ 		plaintext = ciphertext[:len(plaintext)]
+ 	}
+ 
+-	gcm.SealWithRandomNonce(g.GCM, nonce, ciphertext, plaintext, additionalData)
++	if gc, ok := g.aead.(*gcm.GCM); ok {
++		gcm.SealWithRandomNonce(gc, nonce, ciphertext, plaintext, additionalData)
++	} else if gc, ok := g.aead.(randomNonceSealer); ok {
++		gc.SealWithRandomNonce(ciphertext, nonce, plaintext, additionalData)
++	} else {
++		panic("crypto/cipher: GCMWithRandomNonce requires aes.Block or a cipher with random nonce support")
++	}
++
+ 	return ret
+ }
+ 
+@@ -190,7 +204,7 @@ func (g gcmWithRandomNonce) Open(dst, nonce, ciphertext, additionalData []byte)
+ 		ciphertext = ciphertext[gcmStandardNonceSize:]
+ 	}
+ 
+-	_, err := g.GCM.Open(out[:0], nonce, ciphertext, additionalData)
++	_, err := g.aead.Open(out[:0], nonce, ciphertext, additionalData)
+ 	if err != nil {
+ 		return nil, err
+ 	}
+@@ -204,6 +218,10 @@ type gcmAble interface {
+ 	NewGCM(nonceSize, tagSize int) (AEAD, error)
+ }
+ 
++type randomNonceSealer interface {
++	SealWithRandomNonce(out, nonce, plaintext, additionalData []byte)
++}
++
+ func newGCMFallback(cipher Block, nonceSize, tagSize int) (AEAD, error) {
+ 	if tagSize < gcmMinimumTagSize || tagSize > gcmBlockSize {
+ 		return nil, errors.New("cipher: incorrect tag size given to GCM")
+@@ -212,7 +230,16 @@ func newGCMFallback(cipher Block, nonceSize, tagSize int) (AEAD, error) {
+ 		return nil, errors.New("cipher: the nonce can't have zero length")
+ 	}
+ 	if cipher, ok := cipher.(gcmAble); ok {
+-		return cipher.NewGCM(nonceSize, tagSize)
++		g, err := cipher.NewGCM(nonceSize, tagSize)
++		if err != nil {
++			return nil, err
++		}
++
++		if _, ok := g.(randomNonceSealer); ok {
++			return &customGCMWrapper{g}, nil
++		}
++
++		return nil, errors.New("cipher: NewGCMWithRandomNonce requires aes.Block")
```

Comment:
`newGCMFallback` is not used in `NewGCMWithRandomNonce`, so all these changes are unnecessary.